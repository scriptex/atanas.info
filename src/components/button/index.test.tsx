import * as React from 'react';
import { render } from '@testing-library/react';

import { Button } from '.';

// codebeat:disable[ABC,LOC,BLOCK_NESTING]
describe('Button component', () => {
	it('Should render the Button component with type button', () => {
		const { asFragment, rerender } = render(
			<Button type="submit" className="button" onClick={jest.fn()}>
				Button
			</Button>
		);

		expect(asFragment()).toMatchSnapshot();

		rerender(
			<Button type="submit" onClick={jest.fn()}>
				Button
			</Button>
		);

		expect(asFragment()).toMatchSnapshot();

		rerender(
			<Button type="link" className="link" href="https://google.com" rel="noopener noreferrrer" target="_blank">
				Link
			</Button>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
// codebeat:enable[ABC,LOC,BLOCK_NESTING]
