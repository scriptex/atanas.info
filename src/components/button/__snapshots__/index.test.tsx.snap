// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render the Button component with type button 1`] = `
<DocumentFragment>
  <button
    class="c-btn button"
    type="submit"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`Button component Should render the Button component with type button 2`] = `
<DocumentFragment>
  <button
    class="c-btn"
    type="submit"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`Button component Should render the Button component with type button 3`] = `
<DocumentFragment>
  <a
    class="c-btn link"
    href="https://google.com"
    rel="noopener noreferrrer"
    target="_blank"
  >
    Link
  </a>
</DocumentFragment>
`;
